com.hazelcast.jet.JobStatusListenerTest.testListenerDeregistration[onClient] [HZ-2402]

_master_ (commit 88673dbd351713992d538ffe7ff9bcbd3211c34e)

Failed on ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips-nightly/536/testReport/com.hazelcast.jet/JobStatusListenerTest/testListenerDeregistration_onClient_/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<[Jet: STARTING -> RUNNING]> but was:<[Jet: NOT_RUNNING -> STARTING]>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.hazelcast.jet.JobStatusListenerTest.lambda$assertTailEqualsEventually$18(JobStatusListenerTest.java:325)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1277)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1381)
	at com.hazelcast.jet.JobStatusListenerTest.assertTailEqualsEventually(JobStatusListenerTest.java:321)
	at com.hazelcast.jet.JobStatusListenerTest.lambda$testListenerDeregistration$9(JobStatusListenerTest.java:206)
	at com.hazelcast.jet.JobStatusListenerTest.testListener(JobStatusListenerTest.java:307)
	at com.hazelcast.jet.JobStatusListenerTest.testListener(JobStatusListenerTest.java:270)
	at com.hazelcast.jet.JobStatusListenerTest.testListener(JobStatusListenerTest.java:274)
	at com.hazelcast.jet.JobStatusListenerTest.testListenerDeregistration(JobStatusListenerTest.java:200)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:825)

```
</details>


<details><summary>Standard output:</summary>

```
Finished Running Test: testListener_suspendOnFailure[onClient] in 0.222 seconds.
Started Running Test: testListenerDeregistration[onClient]
16:16:15,447 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Creating job classLoader for job 09c5-1eeb-bac4-0002
16:16:15,447 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Create processor classloader map for job 09c5-1eeb-bac4-0002
16:16:15,448 DEBUG || - [Planner] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - Watermarks in the pipeline will be throttled to 1000
16:16:15,448  INFO || - [JobCoordinationService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Starting job 09c5-1eeb-bac4-0002 based on submit request
16:16:15,448 TRACE || - [ClientListenerService] hz.client_72.eventRegistration- - hz.client_72 [dev] [5.3.0-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@eb5c6153, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@69e3bc09} to MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2023-04-27 16:16:15.448, lastWriteTime=2023-04-27 16:16:15.448, closedTime=never}}
16:16:15,449  INFO || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Didn't find any snapshot to restore for job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001
16:16:15,449  INFO || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start executing job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001, execution graph in DOT format:
digraph DAG {
	"streamSource" [localParallelism=1];
	"noop" [localParallelism=1];
	"streamSource" -> "noop" [queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
16:16:15,449 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Building execution plan for job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001
16:16:15,449 TRACE || - [ClientListenerService] hz.client_72.eventRegistration- - hz.client_72 [dev] [5.3.0-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@eb5c6153, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@69e3bc09} to MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2023-04-27 16:16:15.449, lastWriteTime=2023-04-27 16:16:15.449, closedTime=never}}
16:16:15,449 TRACE || - [ClientListenerService] hz.client_72.eventRegistration- - hz.client_72 [dev] [5.3.0-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@eb5c6153, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@69e3bc09} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2023-04-27 16:16:15.445, lastWriteTime=2023-04-27 16:16:15.445, closedTime=never}}
16:16:15,449 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Built execution plans for job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001
16:16:15,449 TRACE || - [ClientListenerService] hz.client_72.eventRegistration- - hz.client_72 [dev] [5.3.0-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.jet.impl.ClientJobProxy$1@eb5c6153, handler=com.hazelcast.jet.impl.ClientJobProxy$ClientJobStatusEventHandler@69e3bc09} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2023-04-27 16:16:15.449, lastWriteTime=2023-04-27 16:16:15.449, closedTime=never}}
16:16:15,449 DEBUG || - [InitExecutionOperation] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09c5-1eeb-bac4-0002, execution 09c5-1ee4-211d-0001 from [127.0.0.1]:5701
16:16:15,450 DEBUG || - [InitExecutionOperation] hz.JobStatusListenerTest_happy_hofstadter.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Initializing execution plan for job 09c5-1eeb-bac4-0002, execution 09c5-1ee4-211d-0001 from [127.0.0.1]:5701
16:16:15,450  INFO || - [JobExecutionService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-15 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution plan for jobId=09c5-1eeb-bac4-0002, jobName='09c5-1eeb-bac4-0002', executionId=09c5-1ee4-211d-0001 initialized
16:16:15,450 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_happy_hofstadter.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Creating job classLoader for job 09c5-1eeb-bac4-0002
16:16:15,450 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_happy_hofstadter.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Create processor classloader map for job 09c5-1eeb-bac4-0002
16:16:15,450  INFO || - [JobExecutionService] hz.JobStatusListenerTest_happy_hofstadter.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution plan for jobId=09c5-1eeb-bac4-0002, jobName='09c5-1eeb-bac4-0002', executionId=09c5-1ee4-211d-0001 initialized
16:16:15,450 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Init of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 was successful
16:16:15,450 DEBUG || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Executing job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001
16:16:15,450  INFO || - [JobExecutionService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 from coordinator [127.0.0.1]:5701
16:16:15,450  INFO || - [JobExecutionService] hz.JobStatusListenerTest_happy_hofstadter.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Start execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 from coordinator [127.0.0.1]:5701
16:16:15,451 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001
16:16:15,451 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 completed
16:16:15,451 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 received response to StartExecutionOperation from [127.0.0.1]:5701: null
16:16:15,452 DEBUG || - [ClientListenerService] hz.client_72.event-1743 - hz.client_72 [dev] [5.3.0-SNAPSHOT] No eventHandler for callId: 125, event: ClientMessage{connection=MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2023-04-27 16:16:15.452, lastWriteTime=2023-04-27 16:16:15.452, closedTime=never}}, length=45, operation=null, isRetryable=false, correlationId=125, messageType=fe1302, isEvent=true, isfragmented=false}
16:16:15,452 DEBUG |testListenerDeregistration[onClient]| - [Job] testListenerDeregistration - hz.client_72 [dev] [5.3.0-SNAPSHOT] Sending CANCEL_FORCEFUL request for job 09c5-1eeb-bac4-0002 (name ??)
16:16:15,453 DEBUG || - [JobClassLoaderService] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=09c5-1eeb-bac4-0002
16:16:15,453 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Completed execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001
16:16:15,453 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5702 [dev] [5.3.0-SNAPSHOT] Execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 completed with failure
java.util.concurrent.CompletionException: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL_FORCEFUL
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:303) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:794) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:761) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:499) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2001) ~[?:1.8.0]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:72) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$ExecutionTracker.taskletDone(TaskletExecutionService.java:498) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.dismissTasklet(TaskletExecutionService.java:429) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:415) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:906) ~[?:1.8.0]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (08-17-2022)]
Caused by: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL_FORCEFUL
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:318) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:723) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution(JobExecutionService.java:719) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.TerminateExecutionOperation.doRun(TerminateExecutionOperation.java:59) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:55) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:192) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:299) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:270) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:502) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:202) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:142) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.loop(OperationThread.java:134) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:115) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:111) ~[classes/:?]
16:16:15,454 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 received response to StartExecutionOperation from [127.0.0.1]:5702: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL_FORCEFUL
16:16:15,454 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-3 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 to be cancelled after Execution
16:16:15,454 DEBUG || - [JobClassLoaderService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=09c5-1eeb-bac4-0002
16:16:15,454  INFO || - [MasterJobContext] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Execution of job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 got terminated, reason=com.hazelcast.jet.impl.exception.CancellationByUserException
	Start time: 2023-04-27T16:16:15.448
	Duration: 00:00:00.006
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
16:16:15,454 DEBUG || - [JobCoordinationService] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] job '09c5-1eeb-bac4-0002', execution 09c5-1ee4-211d-0001 is completed
16:16:15,454  WARN || - [JoinSubmittedJobOperation] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] null
com.hazelcast.jet.impl.exception.CancellationByUserException: null
	at com.hazelcast.jet.impl.MasterJobContext.createCancellationException(MasterJobContext.java:211) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.getErrorFromResponses(MasterJobContext.java:626) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$invokeStartExecution$12(MasterJobContext.java:576) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterContext.lambda$invokeOnParticipant$3(MasterContext.java:361) ~[classes/:?]
	at com.hazelcast.jet.impl.util.ExceptionUtil.lambda$withTryCatch$0(ExceptionUtil.java:180) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture$WhenCompleteNode.lambda$execute$0(AbstractInvocationFuture.java:1570) ~[classes/:?]
	at com.hazelcast.test.TestLoggingUtils$CustomTestNameAwareForkJoinPool$TestNameAwareRunnable.run(TestLoggingUtils.java:104) ~[test-classes/:?]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1413) ~[?:1.8.0]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:300) ~[?:1.8.0]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1078) ~[?:1.8.0]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1714) ~[?:1.8.0]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) ~[?:1.8.0]
16:16:16,835 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:21,835 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:26,835 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:31,835 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-12 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:36,835 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-14 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:41,835 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:46,836 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-14 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:51,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:16:56,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-15 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:01,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-13 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:06,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-13 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:11,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:16,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:21,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-14 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:26,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:31,838 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-13 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:36,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:41,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:46,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:51,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:17:56,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:18:01,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:18:06,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:18:11,839 DEBUG || - [JobRepository] hz.JobStatusListenerTest_trusting_hofstadter.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
16:18:15,608  INFO |testListenerDeregistration[onClient]| - [SimpleTestInClusterSupport] Time-limited test - Ditching 13 jobs in SimpleTestInClusterSupport.@After: [09c5-1ee7-d0c0-0002, 09c5-1ee7-d0c1-0002, 09c5-1ee7-d0c3-0002, 09c5-1ee4-2104-0001, 09c5-1ee4-2106-0001, 09c5-1ee7-d0c7-0002, 09c5-1ee7-d0c8-0002, 09c5-1ee4-2108-0001, 09c5-1eeb-bac4-0002, 09c5-1ee7-d0c9-0002, 09c5-1ee4-210a-0001, 09c5-1ee4-210d-0001, 09c5-1eeb-bac3-0002]
16:18:15,626  INFO |testListenerDeregistration[onClient]| - [SimpleTestInClusterSupport] Time-limited test - Destroying 2 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/advanceCount, hz:impl:mapService/runCount]
BuildInfo right after testListenerDeregistration[onClient](com.hazelcast.jet.JobStatusListenerTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230427', buildNumber=20230427, revision=88673db, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testListenerDeregistration[onClient](com.hazelcast.jet.JobStatusListenerTest):'
16:16:15, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:20, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:25, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:30, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:35, accumulated pauses: 52 ms, max pause: 7 ms, pauses over 1000 ms: 0
16:16:40, accumulated pauses: 47 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:45, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:50, accumulated pauses: 47 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:16:55, accumulated pauses: 47 ms, max pause: 1 ms, pauses over 1000 ms: 0
16:17:00, accumulated pauses: 50 ms, max pause: 1 ms, pauses over 1000 ms: 0
16:17:05, accumulated pauses: 49 ms, max pause: 2 ms, pauses over 1000 ms: 0
16:17:10, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:17:15, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:17:20, accumulated pauses: 47 ms, max pause: 1 ms, pauses over 1000 ms: 0
16:17:25, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:17:30, accumulated pauses: 53 ms, max pause: 3 ms, pauses over 1000 ms: 0
16:17:35, accumulated pauses: 53 ms, max pause: 3 ms, pauses over 1000 ms: 0
16:17:40, accumulated pauses: 43 ms, max pause: 1 ms, pauses over 1000 ms: 0
16:17:45, accumulated pauses: 51 ms, max pause: 2 ms, pauses over 1000 ms: 0
16:17:50, accumulated pauses: 47 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:17:55, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:18:00, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
16:18:05, accumulated pauses: 46 ms, max pause: 1 ms, pauses over 1000 ms: 0
16:18:10, accumulated pauses: 45 ms, max pause: 2 ms, pauses over 1000 ms: 0
16:18:15, accumulated pauses: 9 ms, max pause: 2 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-IbmJDK8-fips-nightly/536/

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlMemoryManagementTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlMemoryManagementTest.java
@@ -29,7 +29,6 @@ import org.junit.experimental.categories.Category;
 
 import java.util.function.Function;
 
-import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
 import static com.hazelcast.sql.impl.type.QueryDataTypeFamily.BIGINT;
 import static com.hazelcast.sql.impl.type.QueryDataTypeFamily.VARCHAR;
 import static java.util.Arrays.asList;
@@ -86,7 +85,7 @@ public class SqlMemoryManagementTest extends SqlTestSupport {
         System.out.println(sql);
         sqlService.execute(sql);
         Job job = instance().getJet().getJob(jobName);
-        assertJobStatusEventually(job, SUSPENDED);
+        assertJobSuspendedEventually(job);
         assertThat(job.getSuspensionCause().errorCause())
                 .contains("Exception thrown to prevent an OutOfMemoryError on this Hazelcast instance");
 

--- a/hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java
@@ -57,7 +57,6 @@ import java.util.function.Function;
 import java.util.function.Supplier;
 
 import static com.hazelcast.client.impl.clientside.ClientTestUtil.getHazelcastClientInstanceImpl;
-import static com.hazelcast.jet.core.JobStatus.RUNNING;
 import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
 import static com.hazelcast.spi.impl.eventservice.impl.EventServiceTest.getEventService;
 import static java.util.Arrays.asList;
@@ -114,11 +113,11 @@ public class JobStatusListenerTest extends SimpleTestInClusterSupport {
     public void testListener_suspend_resume_restart_cancelJob() {
         testListener(streamSource(),
                 (job, listener) -> {
-                    assertJobStatusEventually(job, RUNNING);
+                    assertEqualsEventually(listener::runCount, 1);
                     job.suspend();
                     assertJobStatusEventually(job, SUSPENDED);
                     job.resume();
-                    assertJobStatusEventually(job, RUNNING);
+                    assertEqualsEventually(listener::runCount, 2);
                     job.restart();
                     assertEqualsEventually(listener::runCount, 3);
                     cancelAndJoin(job);
@@ -182,15 +181,13 @@ public class JobStatusListenerTest extends SimpleTestInClusterSupport {
                     throw new JetException("mock error");
                 }),
                 (job, listener) -> {
-                    assertJobStatusEventually(job, SUSPENDED);
-                    String[] failure = new String[1];
-                    assertTrueEventually(() ->
-                            failure[0] = job.getSuspensionCause().errorCause().split("\n", 3)[1]);
+                    assertJobSuspendedEventually(job);
+                    String failure = job.getSuspensionCause().errorCause().split("\n", 3)[1];
                     cancelAndJoin(job);
                     assertTailEqualsEventually(listener.log,
                             "Jet: NOT_RUNNING -> STARTING",
                             "Jet: STARTING -> RUNNING",
-                            "Jet: RUNNING -> SUSPENDED (" + failure[0] + ")",
+                            "Jet: RUNNING -> SUSPENDED (" + failure + ")",
                             "User: SUSPENDED -> FAILED (Cancel)");
                 });
     }
@@ -199,7 +196,9 @@ public class JobStatusListenerTest extends SimpleTestInClusterSupport {
     public void testListenerDeregistration() {
         testListener(streamSource(),
                 (job, listener) -> {
-                    assertJobStatusEventually(job, RUNNING);
+                    assertTailEqualsEventually(listener.log,
+                            "Jet: NOT_RUNNING -> STARTING",
+                            "Jet: STARTING -> RUNNING");
                     listener.deregister();
                     cancelAndJoin(job);
                     assertHasNoListenerEventually(job.getIdString(), listener.registrationId);

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/JetTestSupport.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/JetTestSupport.java
@@ -46,6 +46,7 @@ import com.hazelcast.spi.impl.NodeEngineImpl;
 import com.hazelcast.test.Accessors;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.OverridePropertyRule;
+import org.assertj.core.api.Assertions;
 import org.junit.After;
 import org.junit.ClassRule;
 import org.junit.rules.Timeout;
@@ -70,6 +71,7 @@ import java.util.concurrent.atomic.AtomicReference;
 
 import static com.hazelcast.jet.Util.idToString;
 import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
 import static com.hazelcast.jet.impl.JetServiceBackend.SERVICE_NAME;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
@@ -301,6 +303,20 @@ public abstract class JetTestSupport extends HazelcastTestSupport {
             }
         } while (executionId == null || executionId.equals(ignoredExecutionId));
         return executionId;
+    }
+
+    /**
+     * Asserts that the {@code job} is eventually SUSPENDED and waits until the suspension cause is set.
+     */
+    public static void assertJobSuspendedEventually(Job job) {
+        assertJobStatusEventually(job, SUSPENDED);
+        assertTrueEventually(() -> {
+            try {
+                assertNotNull(job.getSuspensionCause());
+            } catch (IllegalStateException notSuspended) {
+                Assertions.fail("Suspension cause is not set yet", notSuspended);
+            }
+        });
     }
 
     public static void assertJobStatusEventually(Job job, JobStatus expected, int timeoutSeconds) {

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/JobTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/JobTest.java
@@ -1051,7 +1051,7 @@ public class JobTest extends SimpleTestInClusterSupport {
 
         JobConfig config = new JobConfig().setSuspendOnFailure(true).setMaxProcessorAccumulatedRecords(2);
         Job job = instance.getJet().newJob(pipeline, config);
-        assertJobStatusEventually(job, SUSPENDED);
+        assertJobSuspendedEventually(job);
         assertThat(job.getSuspensionCause().errorCause())
                 .contains("Exception thrown to prevent an OutOfMemoryError on this Hazelcast instance");
 

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/SuspendExecutionOnFailureTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/SuspendExecutionOnFailureTest.java
@@ -117,7 +117,7 @@ public class SuspendExecutionOnFailureTest extends TestInClusterSupport {
         Job job = hz().getJet().newJob(dag, jobConfig);
         assertJobStatusEventually(job, RUNNING);
         job.suspend();
-        assertJobStatusEventually(job, SUSPENDED);
+        assertJobSuspendedEventually(job);
         assertThat(job.getSuspensionCause()).matches(JobSuspensionCause::requestedByUser);
         assertThat(job.getSuspensionCause().description()).isEqualTo("Requested by user");
         assertThatThrownBy(job.getSuspensionCause()::errorCause)
@@ -138,7 +138,7 @@ public class SuspendExecutionOnFailureTest extends TestInClusterSupport {
         Job job = hz().getJet().newJob(dag, jobConfig);
 
         // Then
-        assertJobStatusEventually(job, JobStatus.SUSPENDED);
+        assertJobSuspendedEventually(job);
 
         assertThat(job.getSuspensionCause()).matches(JobSuspensionCause::dueToError);
         assertThat(job.getSuspensionCause().errorCause())

--- a/hazelcast/src/test/java/com/hazelcast/jet/impl/memory/MemoryManagementTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/impl/memory/MemoryManagementTest.java
@@ -48,7 +48,6 @@ import java.util.stream.IntStream;
 import static com.hazelcast.function.Functions.wholeItem;
 import static com.hazelcast.jet.Util.entry;
 import static com.hazelcast.jet.aggregate.AggregateOperations.counting;
-import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
 import static com.hazelcast.jet.pipeline.test.AssertionSinks.assertOrdered;
 import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
@@ -137,10 +136,10 @@ public class MemoryManagementTest extends SimpleTestInClusterSupport {
 
         Job job = instance().getJet().newJob(pipeline, new JobConfig().setSuspendOnFailure(true));
         if (expectOOME) {
-            assertJobStatusEventually(job, SUSPENDED);
+            assertJobSuspendedEventually(job);
             assertThat(job.getSuspensionCause().errorCause())
                     .contains("Exception thrown to prevent an OutOfMemoryError on this Hazelcast instance");
-            job.updateConfig(new DeltaJobConfig().setMaxProcessorAccumulatedRecords((long) itemCount));
+            job.updateConfig(new DeltaJobConfig().setMaxProcessorAccumulatedRecords(itemCount));
             job.resume();
         }
         job.join();

